front/lib/api/auth0.ts

```diff
@@ -1,13 +1,38 @@
+import type { Result } from "@dust-tt/types";
+import { Err, Ok } from "@dust-tt/types";
 import { ManagementClient } from "auth0";
+import { isLeft } from "fp-ts/lib/Either";
+import * as t from "io-ts";
 import jwt from "jsonwebtoken";
 import jwksClient from "jwks-rsa";
+import type { NextApiRequest } from "next";
 
 import config from "@app/lib/api/config";
 import { UserResource } from "@app/lib/resources/user_resource";
 import logger from "@app/logger/logger";
 
 let auth0ManagemementClient: ManagementClient | null = null;
 
+const Auth0JwtPayloadSchema = t.type({
+  azp: t.string,
+  exp: t.number,
+  scope: t.string,
+  sub: t.string,
+});
+
+export type Auth0JwtPayload = t.TypeOf<typeof Auth0JwtPayloadSchema> &
+  jwt.JwtPayload;
+
+export function getRequiredScope(
+  req: NextApiRequest,
+  requiredScopes?: Record<string, string>
```

Comment:
imho, the typing of that arg should be more strict (POST|GET|PATCH|DELETE: scopes that exists)

Reply:
req.method is not typed (it's a string), so in theory I can receive anything in here

Reply:
...but in practice we know what we use (and we can extends if needed)

Reply:
i was afraid of the typings, but ok, done 😛